src/awkward/_namedaxis.py

```diff
@@ -0,0 +1,795 @@
+
+    Raises:
+        ValueError: If the named axis is not found in the named axis mapping.
+
+    Examples:
+        >>> _named_axis_to_positional_axis({"x": 0, "y": 1, "z": 2}, "x")
+        0
+    """
+    if _is_valid_named_axis(axis):
+        if axis not in named_axis:
+            raise ValueError(f"{axis=} not found in {named_axis=} mapping.")
+
+        # we wrap it to preserve the original name in its __repr__ and __str__
+        # in order to properly display it in error messages. This is useful for cases
+        # where the positional axis is pointing to a non-existing axis. The error message
+        # will then show the original (named) axis together with the positional axis.
+        cls = _make_named_int_class(axis)
+        return cls(named_axis[axis])
+
+    if is_integer(axis):
+        # TODO: is_integer is an external  helper function that doesn't specify types
+        return int(tp.cast(tp.Any, axis))
+    elif axis is None:
+        return None
+    else:
+        raise ValueError(f"Invalid {axis=} [{type(axis)=}]")
+
+
+# These are the strategies to handle named axis for the
+# output array when performing operations along an axis.
+# See studies/named_axis.md#named-axis-in-high-level-functions and
+# https://pytorch.org/docs/stable/name_inference.html.
+#
+# The possible strategies are:
+# - "keep all" (_keep_named_axis(..., None)): Keep all named axes in the output array, e.g.: `ak.drop_none`
+# - "keep one" (_keep_named_axis(..., int)): Keep one named axes in the output array, e.g.: `ak.firsts`
+# - "keep up to" (_keep_named_axis_up_to(..., int)): Keep all named axes upto a certain positional axis in the output array, e.g.: `ak.local_index`
+# - "remove all" (_remove_all_named_axis): Removes all named axis, e.g.: `ak.categories`
+# - "remove one" (_remove_named_axis): Remove the named axis from the output array, e.g.: `ak.sum`
+# - "add one" (_add_named_axis): Add a new named axis to the output array, e.g.: `ak.concatenate`
+# - "unify" (_unify_named_axis): Unify the named axis in the output array given two input arrays, e.g.: `ak.broadcast_arrays`
+
+
+def _keep_named_axis(
+    named_axis: AxisMapping,
+    axis: int | None = None,
+) -> AxisMapping:
+    """
+    Determines the new named axis after keeping the specified axis. This function is useful when an operation
+    is applied that retains only one axis.
+
+    Args:
+        named_axis (AxisMapping): The current named axis.
+        axis (int | None, optional): The index of the axis to keep. If None, all axes are kept. Default is None.
+
+    Returns:
+        AxisMapping: The new named axis after keeping the specified axis.
+
+    Examples:
+        >>> _keep_named_axis({"x": 0, "y": 1, "z": 2}, 1)
+        {"y": 0}
+        >>> _keep_named_axis({"x": 0, "y": 1, "z": 2}, None)
+        {"x": 0, "y": 1, "z": 2}
+    """
+    if axis is None:
+        return dict(named_axis)
```

Comment:
I'm sure that this copy was deliberate, but I wonder if copying is always needed when it's not required by this function? Is there a way of reducing the copies without making the caller code more complex?